### PR TITLE
Adding Primaza-Identity-Namespace

### PR DESCRIPTION
**Pull request by fbm3307**
Adding Primaza-Identity-Namespace

### CODE REVIEW

File: scripts/src/primazactl/primazaworker/workercluster.py [hunk 1]
```diff
@@ -29,23 +29,24 @@ def __init__(
...
+        super().__init__(context,
                          sa_name,
                          cluster_environment,
                          kubeconfig_file,
                          config_file,
                          cluster_environment,
                          tenant,
-                         internal_url)
+                         internal_url,
+                         pidentity_namespace)
```
Comment (mmulholla) [inline scripts/src/primazactl/primazaworker/workercluster.py:44]: I think pidentity_namespace should be first parameter in place of WORKER_NAMESPACE.

Comment (mmulholla) [inline scripts/src/primazactl/primazaworker/workercluster.py]: when calling super you will need pidentity_namespace to be the first parameter in place of WORKER_NAMESPACE

Comment (fbm3307) [reply scripts/src/primazactl/primazaworker/workercluster.py]: Yes correct, updated it

File: README.md [hunk 2]
```diff
@@ -247,6 +246,9 @@ options:
                         Set for dry run (default: none)
   -o {yaml,none}, --output {yaml,none}
                         Set to get output of resources which are created (default: none).
+  -j PRIMAZA_IDENTITY_NAMESPACE, --primaza_identity_namespace PRIMAZA_IDENTITY_NAMESPACE
```
Comment (filariow) [inline README.md]:
```suggestion
  -j PRIMAZA_IDENTITY_NAMESPACE, --primaza-identity-namespace PRIMAZA_IDENTITY_NAMESPACE
```

File: scripts/src/primazatest/runtest.py [hunk 3]
```diff
@@ -1005,10 +1005,12 @@ def main():
                         dest="input_dir",
                         help="directory for kubeconfigs used for user tests",
                         required=False)
-    parser.add_argument("-j", "--sa_namespace",
-                        dest="sa_namespace",
-                        help="directory for kubeconfigs used for user tests",
-                        required=False)
+    parser.add_argument("-j", "--primaza_identity_namespace",
```
Comment (filariow) [inline scripts/src/primazatest/runtest.py]:
```suggestion
    parser.add_argument("-j", "--primaza-identity-namespace",
```

**Review by filariow — CHANGES_REQUESTED** [review]
some nits

File: scripts/src/primazactl/cmd/create/namespace/common.py [hunk 4]
```diff
@@ -72,6 +73,15 @@ def add_args_namespace(parser: argparse.ArgumentParser, type):
                 on which Primaza's Control Plane is running",
         default=None)
 
+    parser.add_argument(
+        "-j", "--primaza-identity-namespace",
+        dest="primaza_identity_namespace",
+        required=False,
+        help=f"name to be used for the WorkerNamespace that will be created,\
+             default: {WORKER_NAMESPACE}.",
```
Comment (filariow) [inline scripts/src/primazactl/cmd/create/namespace/common.py]:
```suggestion
        help=f"namespace used for hosting the service account shared with\
             Primaza's Control Plane. Default: {WORKER_NAMESPACE}.",
```

File: scripts/src/primazatest/runtest.py (hunk 3 above)
Comment (fbm3307) [reply scripts/src/primazatest/runtest.py]: The _ is there is all the other  options so thought to keep the same way

File: scripts/src/primazactl/cmd/create/namespace/common.py (hunk 4 above)
Comment (fbm3307) [reply scripts/src/primazactl/cmd/create/namespace/common.py:81]: The _ is there is all the other options so thought to keep the same way

File: README.md (hunk 2 above)
Comment (fbm3307) [reply README.md]: The _ is there is all the other options so thought to keep the same way

File: scripts/src/primazactl/cmd/apply/cluster_environment.py [hunk 5]
```diff
@@ -47,13 +48,22 @@ def __init__(self, options, tenant):
         self.version = tenant.version if tenant.version \
             else defaults["version"]
 
+        self.service_account_namespace = \
+            options.get("primazaIdentityNamespace", None)
```
Comment (mmulholla) [inline scripts/src/primazactl/cmd/apply/cluster_environment.py]: Need to change this to new name.

File: scripts/src/primazatest/runtest.py [hunk 6]
```diff
@@ -1005,10 +1005,12 @@ def main():
                         dest="input_dir",
                         help="directory for kubeconfigs used for user tests",
                         required=False)
-    parser.add_argument("-j", "--sa_namespace",
-                        dest="sa_namespace",
-                        help="directory for kubeconfigs used for user tests",
-                        required=False)
+    parser.add_argument("-j", "--service_account_namespace",
```
Comment (mmulholla) [inline scripts/src/primazatest/runtest.py]: Need to now use this to pass as a parameter in one of the tests, so tests include running with a named namespace.
Also makefile needs to be updated to pass the parameter to runtest.

**Review by mmulholla — CHANGES_REQUESTED** [review]

File: scripts/src/primazatest/runtest.py (hunk 6 above)
Comment (filariow) [reply scripts/src/primazatest/runtest.py]: shouldn't it be `--service-account-namespace`?

File: README.md [hunk 7]
```diff
@@ -247,6 +246,9 @@ options:
                         Set for dry run (default: none)
   -o {yaml,none}, --output {yaml,none}
                         Set to get output of resources which are created (default: none).
+  -j SERVICE_ACCOUNT_NAMESPACE, --service_account_namespace SERVICE_ACCOUNT_NAMESPACE
```
Comment (filariow) [inline README.md]: should this be `service-account-namespace`?
```suggestion
  -j SERVICE_ACCOUNT_NAMESPACE, --service-account-namespace SERVICE_ACCOUNT_NAMESPACE
```

File: scripts/src/primazactl/cmd/create/namespace/common.py (hunk 4 above)
Comment (fbm3307) [reply scripts/src/primazactl/cmd/create/namespace/common.py]: done

File: scripts/src/primazactl/cmd/apply/cluster_environment.py (hunk 5 above)
Comment (fbm3307) [reply scripts/src/primazactl/cmd/apply/cluster_environment.py]: done

File: README.md (hunk 7 above)
Comment (fbm3307) [reply README.md]: yes good catch on this, updated it

File: scripts/src/primazatest/runtest.py (hunk 6 above)
Comment (fbm3307) [reply scripts/src/primazatest/runtest.py]: yes good catch on this, updated it

**Review by mmulholla — DISMISSED** [review]

File: scripts/src/primazactl/cmd/create/namespace/common.py [hunk 8]
```diff
@@ -72,6 +73,16 @@ def add_args_namespace(parser: argparse.ArgumentParser, type):
                 on which Primaza's Control Plane is running",
         default=None)
 
+    parser.add_argument(
+        "-j", "--service-account-namespace",
+        dest="service_account_namespace",
+        required=False,
+        help=f"namespace used for hosting the service account shared with\
+             Primaza's Control Plane(existing namespace).\
```
Comment (filariow) [inline scripts/src/primazactl/cmd/create/namespace/common.py:81]:
```suggestion
        help=f"namespace used for hosting the service account shared with\
             Primaza's Control Plane. The namespace should be already existing. \
```

File: scripts/src/primazactl/cmd/create/namespace/common.py [hunk 9]
```diff
@@ -72,6 +73,16 @@ def add_args_namespace(parser: argparse.ArgumentParser, type):
...
+             Default: {WORKER_NAMESPACE}.",
+        type=kubernetes_name,
+        default=None)
+
```
Comment (filariow) [inline scripts/src/primazactl/cmd/create/namespace/common.py]: This value should be required only in the `join cluster` command. not in create `create *-namespace`.

File: scripts/src/primazactl/cmd/join/parser.py [hunk 10]
```diff
@@ -86,6 +87,16 @@ def add_args_join(parser: argparse.ArgumentParser):
         type=str,
         default=None)
 
+    parser.add_argument(
+        "-j", "--service-account-namespace",
+        dest="service_account_namespace",
+        required=False,
+        help=f"namespace used for hosting the service account shared with\
+             Primaza's Control Plane(existing namespace).\
```
Comment (filariow) [inline scripts/src/primazactl/cmd/join/parser.py]:
```suggestion
        help=f"namespace used for hosting the service account shared with\
             Primaza's Control Plane. The namespace should be already existing. \
```

File: scripts/src/primazatest/options/primaza-alice.yaml [hunk 11]
```diff
@@ -1,21 +1,21 @@
...
     internalUrl: https://172.18.0.3:6443
     kubeconfig: ~/.kube/config
-  serviceNamespaces:
-    - name: alice-svc
-  applicationNamespaces:
-    - name: alice-app
-
+controlPlane:
+  context: kind-primazactl-tenant-test
+  internalUrl: https://172.18.0.2:6443
```
Comment (filariow) [inline scripts/src/primazatest/options/primaza-alice.yaml]: this can introduce some problems, should we revert changes to this file?

Comment (fbm3307) [reply scripts/src/primazatest/options/primaza-alice.yaml]: will do

File: scripts/src/primazactl/cmd/create/namespace/common.py (hunk 9 above)
Comment (fbm3307) [reply scripts/src/primazactl/cmd/create/namespace/common.py]: this is required because the method is used for  creating cluster environment

Comment (filariow) [reply scripts/src/primazactl/cmd/create/namespace/common.py]: I can see, you're right

**Review by filariow — APPROVED** [review]
lgtm

**Pull request merged**